internal/magic/text.go

```diff
@@ -379,3 +379,29 @@ func scanLine(b []byte) (line, remainder []byte) {
 	line, remainder, _ = bytes.Cut(b, []byte("\n"))
 	return dropCR(line), remainder
 }
+
+// Gltf matches a GL Transmission Format (JSON) file
+// Visit [glTF specification] and [IANA glTF entry] for more details.
+//
+// [glTF specification]: https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html
+// [IANA glTF (JSON) entry]: https://www.iana.org/assignments/media-types/model/gltf+json
+func Gltf(raw []byte, limit uint32) bool {
+	raw = trimLWS(raw)
+	if len(raw) == 0 || raw[0] != '{' {
+		return false
+	}
+
+	// Check for required GLTF identifiers
+	gltfIdentifiers := [][]byte{
+		[]byte(`"asset"`),
+		[]byte(`"version"`),
+	}
+
+	for _, id := range gltfIdentifiers {
+		if !bytes.Contains(raw, id) {
```

Comment:
This check is not reliable.

It was a problem present in other detectors as well (example: [geojson detector](https://github.com/gabriel-vasile/mimetype/blob/68c306d8f90801870af022c9694a501549cf3927/internal/magic/text.go#L187-L188))

But I have added a new json parser that can do these checks reliable; see #655.